extensions/amp-audio/1.0/test/test-amp-audio.js

```diff
@@ -0,0 +1,378 @@
+              childNodeAttrs.tag,
+              childNodeAttrs
+            );
+          }
+          ampAudio.appendChild(child);
+        });
+      }
+      doc.body.appendChild(ampAudio);
+      return ampAudio;
+    }
+
+    /**
+     * Creates `<amp-audio>`
+     * @param {`*`} attributes Attributes to be add
+     * @param {*} opt_childNodesAttrs Child nodes to be add
+     * @returns Returns `<amp-audio>` with given parameters
+     */
+    function attachAndRun(attributes, opt_childNodesAttrs) {
+      naturalDimensions_['AMP-AUDIO'] = {width: '300px', height: '30px'};
+      const ampAudio = getAmpAudio(attributes, opt_childNodesAttrs);
+      return ampAudio;
+    }
+
+    /**
+     * Creates `<amp-story>` and add `<amp-audio>` as child
+     * @param {*} attributes Attributes to be add
+     * @returns Returns `<amp-story>` with `<amp-audio>` as child
+     */
+    function attachToAmpStoryAndRun(attributes) {
+      naturalDimensions_['AMP-AUDIO'] = {width: '300px', height: '30px'};
+      const ampAudio = doc.createElement('amp-audio');
+      const ampStory = doc.createElement('amp-story');
+      for (const key in attributes) {
+        ampAudio.setAttribute(key, attributes[key]);
+      }
+      ampStory.appendChild(ampAudio);
+      doc.body.appendChild(ampStory);
+
+      return ampAudio;
+    }
+
+    it('should load audio through attribute', async () => {
+      element = attachAndRun({
+        src: 'audio.mp3',
+      });
+
+      // Wait till rendering is finished
+      await waitForRender();
+
+      // Retrieve
+      const {shadowRoot} = element;
+      const audio = shadowRoot.querySelector('audio');
+
+      expect(audio.tagName).to.equal('AUDIO');
+      expect(audio.getAttribute('src')).to.equal('audio.mp3');
+      expect(audio.hasAttribute('controls')).to.be.true;
+      expect(element.style.width).to.be.equal('300px');
+      expect(element.style.height).to.be.equal('30px');
+    });
+
+    it('should not preload audio', async () => {
+      element = attachAndRun({
+        src: 'audio.mp3',
+        preload: 'none',
+      });
+
+      // Wait till rendering is finished
+      await waitForRender();
+
+      // Fetch audio element from shadowRoot
+      const {shadowRoot} = element;
+      const audio = shadowRoot.querySelector('audio');
+
+      expect(audio.getAttribute('preload')).to.be.equal('none');
+    });
+
+    it('should only preload audio metadata', async () => {
+      element = attachAndRun({
+        src: 'audio.mp3',
+        preload: 'metadata',
+      });
+
+      // Wait till rendering is finished
+      await waitForRender();
+
+      // Fetch audio element from shadowRoot
+      const {shadowRoot} = element;
+      const audio = shadowRoot.querySelector('audio');
+
+      expect(audio.getAttribute('preload')).to.be.equal('metadata');
+    });
+
+    it(
+      'should attach `<audio>` element and execute relevant actions for ' +
+        'layout="nodisplay"',
+      async () => {
```

Comment:
nit: This might have been inherited, but it's nicer to keep the test name in one line, since it prevents the body from going one level deeper (the formatter knows that it should keep the initial line unbroken.

```suggestion
    it('should attach `<audio>` element and execute relevant actions for layout="nodisplay"', async () => {
```